Formula/grokj2k.rb

```diff
@@ -22,36 +22,128 @@ class Grokj2k < Formula
 
   depends_on "cmake" => :build
   depends_on "doxygen" => :build
+  depends_on "pkg-config" => :build
   depends_on "exiftool"
   depends_on "jpeg-turbo"
   depends_on "libpng"
   depends_on "libtiff"
   depends_on "little-cms2"
 
+  uses_from_macos "perl"
+  uses_from_macos "zlib"
+
+  on_macos do
+    # HACK: this should not be a test dependency but is due to a limitation with fails_with
+    depends_on "llvm" => [:build, :test] if DevelopmentTools.clang_build_version >= 1205
+  end
+
+  on_linux do
+    depends_on "gcc"
+  end
+
+  # https://github.com/GrokImageCompression/grok/blob/master/INSTALL.md#compilers
+  fails_with :gcc do
+    version "9"
+    cause "GNU compiler version must be at least 10.0"
+  end
+
+  # Build failed with Apple clang version 12.0.5 and 13.0.0.
+  # clang: error: unable to execute command: Segmentation fault: 11
+  # clang: error: clang frontend command failed due to signal (use -v to see invocation)
+  # Upstream issue closed as Apple clang bug: https://github.com/GrokImageCompression/grok/issues/256
+  fails_with :clang if DevelopmentTools.clang_build_version >= 1205
+
+  resource "test_image" do
+    url "https://raw.githubusercontent.com/GrokImageCompression/input_image_test_suite/173de0ae73371751f857d16fdaf2c3301e54a3a6/exif-samples/tiff/Tless0.tiff"
+    sha256 "32f6aab90dc2d284a83040debe379e01333107b83a98c1aa2e6dabf56790b48a"
+  end
+
   def install
-    system "cmake", ".", *std_cmake_args, "-DBUILD_DOC=ON"
-    system "make", "install"
+    args = ["-DBUILD_DOC=ON"]
+
+    if DevelopmentTools.clang_build_version >= 1205
+      ENV.remove "HOMEBREW_LIBRARY_PATHS", Formula["llvm"].opt_lib
+      ENV.llvm_clang
+      # Workaround issue with LLVM install_name_tool, which causes errors during install.
+      # .../llvm/bin/install_name_tool: error: unsupported load command (cmd=0x80000034)
+      args << "-DCMAKE_INSTALL_NAME_TOOL=/usr/bin/install_name_tool"
```

Comment:
@carlocab This workaround may be related to #95319. LLVM 13's `install_name_tool` is causing issues for ARM when modifying RPATH.

Reply:
Yea, we should revert that part